client/v2/algod/block.go

```diff
@@ -0,0 +1,34 @@
+package algod
+
+import (
+	"context"
+	"fmt"
+	"github.com/algorand/go-algorand-sdk/client/v2/common"
+	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+type Block struct {
+	c     *Client
+	round uint64
+	p     models.GetBlockParams
+}
+
+type generatedBlockResponse struct {
+	// Block header data.
+	Block types.Block `json:"block"`
```

Comment:
Is `types.Block` the block object from the indexer endpoint? If so, that isn't the right type. This should also be `map[string]interface{}`

Reply:
It's not, it's the `codec`-encoded/msgpack-represented block from `go-algorand-sdk/types`